linesearch.go

```diff
@@ -62,6 +62,11 @@ func (ls *Linesearch) Iterate(loc *Location, xNext []float64) (EvaluationType, I
 		return NoEvaluation, ls.iterType, nil
 	}
 	if ls.iterType == MajorIteration {
+		// Compare the starting points for the previous and next iteration to
```

Comment:
In fact, initially I had the test after the floats.AddScaledTo because that seemed like the obvious right place. But it caused some failures in the test suite, so I thought that it was not so obvious, because without this new check the tests completed fine. I will look into it. I suspect that the explanation is the following: Backtracking produces monotonically decreasing sequence of step sizes, so once x==x+step*dir, it will not recover. However, Bisection can increase the step size, so x==x+step*dir does not mean that it cannot recover on the next minor iteration. Apparently, it can. So after floats.AddScaledTo does not seem like the right place either.